fetch_response.go

```diff
@@ -79,15 +82,51 @@ func (b *FetchResponseBlock) decode(pd packetDecoder, version int16) (err error)
 	if err != nil {
 		return err
 	}
-	if recordsSize > 0 {
-		if err = b.Records.decode(recordsDecoder); err != nil {
+
+	b.RecordsSet = []*Records{}
+
+	for {
+		if recordsDecoder.remaining() == 0 {
```

Comment:
I think `for recordsDecoder.remaining() > 0 {` would be more idiomatic

Reply:
That would require one more level of nesting, I think it's better to break early and keep it flat:

* https://softwareengineering.stackexchange.com/a/18459

Reply:
I'm not sure I follow; moving the condition into the `for` loop that already exists wouldn't add any layers of nesting?

Reply:
My bad, I misread the first time. Code is changed now.